sandbox/lua/encoders/schema_influx_write.lua

```diff
@@ -281,27 +277,11 @@ function process_message()
         -- Exit the perpetual loop when the iteration is complete
         if not typ then break end
 
-        -- Only process fields that are not requested to be skipped
-        if not skip_fields_str or not skip_fields[name] then
-            -- Set the name attribute of this table by concatenating name_prefix
-            -- with the name of this particular field
-            local field_name = name_prefix.."."..name
-
-            -- Merge existing base fields with field in this iteration
-            local fields = {}
-            for k,v in pairs(columns) do fields[v] = values[k] end
-
-            -- Structure the table to match the expected Influxdb structure
-            fields["value"] = value
-            points[points_index] = {
-                name = field_name,
-                fields = fields
-            }
-
-            -- Increment the index to the table so the next iteration will
-            -- append another entry to the array. This allows us to send a
-            -- bunch of metrics to InfluxDB with a single HTTP request
-            points_index = points_index + 1
+        -- Force to remain as a true float if value is 0; otherwise
+        -- InfluxDB will reject the point if previous values were
+        -- actual floats
+        if value == 0 and typ == 3 then
```

Comment:
@davidbirdsong It's pretty ugly, but the workaround for lua converting 0.0 to 0 is here and then followed up on line 334.  It basically just converts it to a string and then removes the quotes when it's concatenating values when injecting a new message with payload composed of one or more series that are newline separated.